src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -257,6 +258,338 @@ private struct IntPtrAndHandle
+                case TypeCode.Int16:
+                case TypeCode.UInt16:
+                case TypeCode.Int32:
+                    if (type.IsEnum)
+                        return MarshalType.ENUM;
+                    else
+                        return MarshalType.INT;
+                case TypeCode.UInt32:
+                    if (type.IsEnum)
+                        return MarshalType.ENUM;
+                    else
+                        return MarshalType.UINT32;
+                case TypeCode.Boolean:
+                    return MarshalType.BOOL;
+                case TypeCode.Int64:
+                    if (type.IsEnum)
+                        return MarshalType.ENUM64;
+                    else
+                        return MarshalType.INT64;
+                case TypeCode.UInt64:
+                    if (type.IsEnum)
+                        return MarshalType.ENUM64;
+                    else
+                        return MarshalType.UINT64;
+                case TypeCode.Single:
+                    return MarshalType.FP32;
+                case TypeCode.Double:
+                    return MarshalType.FP64;
+                case TypeCode.String:
+                    return MarshalType.STRING;
+                case TypeCode.Char:
+                    return MarshalType.CHAR;
+            }
+
+            if (typeof(Array).IsAssignableFrom(type)) {
+                switch (Type.GetTypeCode(type.GetElementType())) {
+                    case TypeCode.Byte:
+                        return MarshalType.ARRAY_UBYTE;
+                    case TypeCode.SByte:
+                        return MarshalType.ARRAY_BYTE;
+                    case TypeCode.Int16:
+                        return MarshalType.ARRAY_SHORT;
+                    case TypeCode.UInt16:
+                        return MarshalType.ARRAY_USHORT;
+                    case TypeCode.Int32:
+                        return MarshalType.ARRAY_INT;
+                    case TypeCode.UInt32:
+                        return MarshalType.ARRAY_UINT;
+                    case TypeCode.Single:
+                        return MarshalType.ARRAY_FLOAT;
+                    case TypeCode.Double:
+                        return MarshalType.ARRAY_DOUBLE;
+                    case TypeCode.Char:
+                    case TypeCode.Boolean:
+                    case TypeCode.Int64:
+                    case TypeCode.UInt64:
+                        // FIXME
+                        return MarshalType.OBJECT;
+                    default:
+                        // FIXME
+                        return MarshalType.OBJECT;
+                }
+            } else if (type == typeof(void))
+                return MarshalType.VOID;
+            else if (type == typeof(IntPtr))
+                return MarshalType.INT;
+            else if (type == typeof(UIntPtr))
+                return MarshalType.UINT32;
+            else if (typeof(Delegate).IsAssignableFrom(type))
```

Comment:
```type == typeof(Delegate)``` will never work, will it? And ```type is Delegate``` won't work because ```type is Type```. Am I missing a better BCL method I could be using for this

Reply:
ok, missed the details that it's Type